src/bin/cargo/commands/fix.rs

```diff
@@ -72,6 +72,20 @@ pub fn cli() -> App {
                 .long("allow-staged")
                 .help("Fix code even if the working directory has staged changes"),
         )
+        .arg(
+            Arg::with_name("clippy")
+                .long("clippy")
+                .help("Get fix suggestions from clippy instead of check")
+                .hidden(true),
+        )
+        .arg(
+            Arg::with_name("clippy-arg")
```

Comment:
I'm not quite convinced having a separate argument is necessary.  Currently clippy-driver takes very few arguments, and their format is quite simple.  I think just allowing `--clippy` to take an optional string of arguments which can be split on spaces would be sufficient.

Reply:
I don't know how to get clap to allow us to either pass values or not pass values. If I don't put `takes_value(true)` on the arg it will always yield `None` when I try to extract the value passed in, but if I don't specify that it takes a value it will error if I try to invoke `--clippy` without any additional arguments.

Reply:
[`optional_multi_opt`](https://github.com/rust-lang/cargo/blob/143aaed0eedda82f6684637ba1bf2d69b24c7799/src/cargo/util/command_prelude.rs#L204-L214) contains the incantation to have an optional value.  Call `is_present` to see if it is set, and `value_of` to get the actual value.